File: TesteAndroid/app/src/main/java/com/accenture/android/app/testeandroid/presentation/main/adapters/StatementsRecyclerAdapter.java
package com.accenture.android.app.testeandroid.presentation.main.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.accenture.android.app.testeandroid.databinding.ItemRecyclerStatementBinding;
import com.accenture.android.app.testeandroid.domain.Statement;
import com.accenture.android.app.testeandroid.helpers.FormatHelper;

import java.text.ParseException;
import java.util.ArrayList;

/**
 * Created by Denis Magno on 09/07/2020.
 * denis_magno16@hotmail.com
 */
public class StatementsRecyclerAdapter extends RecyclerView.Adapter<StatementsRecyclerAdapter.StatementViewHolder> {
    private final static String TAG = "CustomLog - " + StatementsRecyclerAdapter.class.getSimpleName();

    private final Context context;

    private ArrayList<Statement> statements;

    public StatementsRecyclerAdapter(Context context, ArrayList<Statement> statements) {
        this.context = context;
        this.statements = statements;
    }

    @NonNull
    @Override
    public StatementsRecyclerAdapter.StatementViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = (LayoutInflater) this.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        ItemRecyclerStatementBinding binding = ItemRecyclerStatementBinding.inflate(layoutInflater, parent, false);

        return new StatementViewHolder(this.context, binding);
    }

    @Override
    public void onBindViewHolder(@NonNull StatementsRecyclerAdapter.StatementViewHolder holder, int position) {
        holder.bindStatement(this.statements.get(position));
    }

    @Override
    public int getItemCount() {
        return this.statements.size();
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    public static class StatementViewHolder extends RecyclerView.ViewHolder {
        private final static String TAG = "CustomLog - " + StatementViewHolder.class.getSimpleName();

        private Context context;
        private ItemRecyclerStatementBinding binding;

        public StatementViewHolder(Context context, ItemRecyclerStatementBinding binding) {
            super(binding.getRoot());

            this.context = context;
            this.binding = binding;
        }

        public void bindStatement(Statement statement) {
            this.binding.txtTitulo.setText(statement.getTitle());
            this.binding.txtDescricao.setText(statement.getDesc());

            try {
                this.binding.txtData.setText(FormatHelper.formatarData(statement.getDateCalendar()));
            } catch (ParseException e) {
                e.printStackTrace();
            }

            this.binding.txtValor.setText(FormatHelper.formatarReal(statement.getValue()));
        }
    }
}

File: BankApp/app/src/main/java/com/accenture/bankapp/screens/dashboard/StatementsRecyclerAdapter.kt
package com.accenture.bankapp.screens.dashboard

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.accenture.bankapp.R
import com.accenture.bankapp.network.models.Statement
import timber.log.Timber
import java.text.MessageFormat
import java.text.NumberFormat
import java.util.*

internal class StatementViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val textTitle: TextView = itemView.findViewById(R.id.textTitle)
    val textDate: TextView = itemView.findViewById(R.id.textDate)
    val textDesc: TextView = itemView.findViewById(R.id.textDesc)
    val textValue: TextView = itemView.findViewById<TextView>(R.id.textValue)
}

class StatementsRecyclerAdapter(
    private var context: Context,
    private var listStatements:  MutableList<Statement>
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        Timber.i("onCreateViewHolder: Creating View Holder")

        return StatementViewHolder(
            LayoutInflater.from(context).inflate(
                R.layout.card_statement,
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is StatementViewHolder) {
            val statement = listStatements[position]

            Timber.i("onBindViewHolder: Binding Holder $holder with Statement $statement")

            val formattedDate = MessageFormat("{2}/{1}/{0}").format(statement.date.split("-").toTypedArray())
            val formattedValue = NumberFormat.getCurrencyInstance(Locale("pt", "BR")).format(statement.value)

            holder.textTitle.text = statement.title
            holder.textDate.text = formattedDate
            holder.textDesc.text = statement.desc
            holder.textValue.text = formattedValue
        }
    }

    override fun getItemCount(): Int {
        return listStatements.size
    }
}